### DISCUSSION TITLE
[Matter] expose S/N for Matter devices in HA webUI

### DISCUSSION THREAD

**Discussion by lboue** [home-assistant/frontend#18491] [Other feature requests] [answered]
Hello,

Do you think we could expose S/N for Matter devices in HA webUI ?
It would be much easier to distinguish device of the same type connected via a Matter bridge

Regards

**Comment by lboue** [comment]
Do you know why the value this.device.serial_number is not displayed, even though the following code is present?
[ha-device-info-card.ts](https://github.com/home-assistant/frontend/blob/3ac7cd5d4acd0ec32d514a03a9af32e3e81eedcd/src/panels/config/devices/device-detail/ha-device-info-card.ts#L86C25-L86C38)


```js
          ${this.device.serial_number
            ? html`
                <div class="extra-info">
                  ${this.hass.localize(
                    "ui.panel.config.integrations.config_entry.serial_number",
                    { serial_number: this.device.serial_number }
                  )}
                </div>
              `
            : ""}
```



Is it simply because it is not instantiated on the Matter server side?


**Data from diag file**
```json
{
  "28/29/0": [
    {
      "deviceType": 19,
      "revision": 1
    },
    {
      "deviceType": 17,
      "revision": 1
    }
  ],
  "28/29/1": [
    29,
    57,
    46,
    47
  ],
  "28/29/2": [],
  "28/29/3": [
    29,
    30,
    31,
    32,
    33,
    34
  ],
  "28/29/65532": 0,
  "28/29/65533": 1,
  "28/29/65528": [],
  "28/29/65529": [],
  "28/29/65531": [
    0,
    1,
    2,
    3,
    65528,
    65529,
    65531,
    65532,
    65533
  ],
  "28/57/1": "Aqara",
  "28/57/2": 4447,
  "28/57/3": "Aqara Cube T1 Pro",
  "28/57/5": "Aqara Cube T1 Pro",
  "28/57/7": 1,
  "28/57/8": "1",
  "28/57/9": 25,
  "28/57/10": "25",
  "28/57/15": "54ef4410006a3881",
  "28/57/12": "AR013",
  "28/57/13": "https://www.aqara.com/en/products.html",
  "28/57/14": "Aqara Cube T1 Pro",
  "28/57/17": true,
  "28/57/18": "lumi.54ef4410006a3881",
  "28/57/65532": 0,
  "28/57/65533": 1,
  "28/57/65528": [],
  "28/57/65529": [],
  "28/57/65531": [
    1,
    2,
    3,
    5,
    7,
    8,
    9,
    10,
    15,
    12,
    13,
    14,
    17,
    18,
    65528,
    65529,
    65531,
    65532,
    65533
  ],
  "28/46/0": [
    28
  ],
  "28/46/65532": 0,
  "28/46/65533": 1,
  "28/46/65528": [],
  "28/46/65529": [],
  "28/46/65531": [
    0,
    65528,
    65529,
    65531,
    65532,
    65533
  ],
  "28/47/0": 1,
  "28/47/1": 0,
  "28/47/2": "Battery",
  "28/47/11": 3000,
  "28/47/12": 200,
  "28/47/14": 0,
  "28/47/15": false,
  "28/47/16": 2,
  "28/47/19": "CR2032",
  "28/47/25": 1,
  "28/47/65532": 10,
  "28/47/65533": 1,
  "28/47/65528": [],
  "28/47/65529": [],
  "28/47/65531": [
    0,
    1,
    2,
    11,
    12,
    14,
    15,
    16,
    19,
    25,
    65528,
    65529,
    65531,
    65532,
    65533
  ]
}
```

**Comment by lboue** [comment]
May I have an answer to my question?

**Comment by matthiasdebaat** [comment]
Currently we're working on showing more details in a collapsable like this Zigbee example. Can't give you an ETA.

![CleanShot 2024-01-23 at 20 36 55@2x](https://github.com/home-assistant/frontend/assets/20203284/c53a5492-6694-4601-984d-22ce27927cf2)

**Comment by lboue** [comment]
@matthiasdebaat 
Here's what I currently get with HA last version for Eve Energy Thread plug:

**Device info**

- Eve Energy 20EBO8301
- by Eve Systems
- Firmware: 3.2.1
- Hardware: 1.1

**Device info**
- Node ID: 3
- Network Type: Thread
- Device type: Routing end device
- Network name: MyHome36
- MAC address: 52:3f:32:db:15:3a:41:85
- IP Address(es): fda9:150:8b20:0:c697:e330:6b46:8d29

I can't find the serial number.

![image](https://github.com/home-assistant/frontend/assets/938089/b1264469-2a40-41f4-ba66-0f986a2a08dc)

**Reply by marcelveldt** [reply, marked as answer]
we'll add the missing serial number in one of the future updates. we simply forgot it in the initial implementation.
